python/ray/train/lightning/lightning_trainer.py

```diff
@@ -373,7 +385,14 @@ def __init__(
         run_config = run_config or RunConfig()
         lightning_config = lightning_config or LightningConfigBuilder().build()
 
-        self._check_checkpoint_configs(
+        if datasets and not datasets_iter_config:
+            raise RuntimeError(
+                "No `datasets_iter_config` provided for the input `datasets`!"
+                "Please refer to the API of `ray.data.Dataset.iter_torch_batches`"
+                "for all valid arguments."
+            )
+
+        run_config.checkpoint_config = self._unify_checkpoint_configs(
```

Comment:
We are modifying the user's run config. Should we instead make a copy then use that internally in the LightningTrainer?